Hole05

Julesc002 opened the pull request.

Commit:
hole 3

Commit:
enable linter

Prof-Iter committed:
Tackle scope of constants and variables

Prof-Iter committed:
Tackle feature envy

Prof-Iter committed:
meilleures explications avec exemples

D4rkP4ng0l1n commented:
Modification de l'interface Tile. Désormais Tile sera une classe qui servira à tester les différentes conditions des if de la classe Game. Grâce à la classe Tile, les conditions if comprenant un this._board.TileAt verront leur écriture simplifié dans la classe Game.

D4rkP4ng0l1n commented on src/12_RefactoringGolf/hole1/kata.ts:
```diff
@@ -1,78 +1,86 @@
+  }
+
+  private validatePositionIsEmpty(x: number, y: number) {
+    if (this._board.TileAt(x, y).Symbol != emptyPlay) {
```
Création d'une méthode isNotEmpty dans la classe Tile

D4rkP4ng0l1n commented on src/12_RefactoringGolf/hole1/kata.ts:
```diff
@@ -1,78 +1,86 @@
+    return (
+      this._board.TileAt(row, firstColumn)!.Symbol != emptyPlay &&
+      this._board.TileAt(row, secondColumn)!.Symbol != emptyPlay &&
+      this._board.TileAt(row, thirdColumn)!.Symbol != emptyPlay
```
Utilisation de la méthode isNotEmpty pour les 3 conditions du return

D4rkP4ng0l1n commented on src/12_RefactoringGolf/hole1/kata.ts:
```diff
@@ -1,78 +1,86 @@
+      this._board.TileAt(row, firstColumn)!.Symbol ==
+        this._board.TileAt(row, secondColumn)!.Symbol &&
+      this._board.TileAt(row, thirdColumn)!.Symbol == this._board.TileAt(row, secondColumn)!.Symbol
+    );
```
Création d'une méthode hasSameSymbolAs dans la classe Tile

D4rkP4ng0l1n commented on src/12_RefactoringGolf/hole1/kata.ts:
```diff
@@ -99,9 +107,6 @@ class Board {
-    //@ts-ignore
-    const tile: Tile = { X: x, Y: y, Symbol: symbol };
-
     this._plays.find((t: Tile) => t.X == x && t.Y == y)!.Symbol = symbol;
```
Création d'une méthode hasSameCoordinatesAs et UpdateSymbol dans la classe Tile

Julesc002 committed:
test

Julesc002 committed:
Modification de l'interface Tile

+ Création d'une méthode isNotEmpty dans la classe Tile
+ Utilisation de la méthode isNotEmpty pour les 3 conditions du return
+ Création d'une méthode hasSameSymbolAs dans la classe Tile
+ Création d'une méthode hasSameCoordinatesAs et UpdateSymbol dans la classe Tile
